templates/github/.github/workflows/scripts/update_backport_labels.py.j2

```diff
@@ -0,0 +1,37 @@
+import requests
+import json
+import yaml
+import random
+
+def random_color():
+    color = random.randrange(0, 2**24)
+    return format(color, '06x')  
+
+token = "placeholder"
+
+headers = { "Authorization": f"token {token}",
+            "Accept": "application/vnd.github.v3+json",
+            "X-GitHub-Api-Version": "2022-11-28" }
+
+# list all labels from the repository's current state
+response = requests.get("https://api.github.com/repos/pulp/pulpcore/labels", headers=headers)
+assert(response.status_code == 200)
+old_labels = [x["name"] for x in response.json() if x["name"].startswith("backport-")]
+
+# list ci_update_branches from template_config.yml
+plugin_template = yaml.safe_load(open("Pulp/pulpcore/template_config.yml", "r"))
+new_labels = ["backport-" + x for x in plugin_template["ci_update_branches"]]
+
+# delete old labels that are not in new labels
+for label in [l for l in old_labels if l not in new_labels]:
+    response = requests.delete(f"https://api.github.com/repos/pulp/pulpcore/labels/{label}", headers=headers)
+    assert(response.status_code == 204)
+
+# create new labels that are not in old labels
+for label in [l for l in new_labels if l not in old_labels]:
+    color = random_color()
+    response = requests.post("https://api.github.com/repos/pulp/pulpcore/labels", headers=headers, data=json.dumps({"name": label, "color": color}))
```

Comment:
I'm pretty confident, `requests` can handle json for us. Also consider using a session.